src/image/BackupIncrements.cc

```diff
@@ -0,0 +1,193 @@
+/* limitations under the License.                                             */
+/* -------------------------------------------------------------------------- */
+
+#include "BackupIncrements.h"
+
+/* -------------------------------------------------------------------------- */
+/* -------------------------------------------------------------------------- */
+
+int BackupIncrements::from_xml_node(const xmlNodePtr node)
+{
+    int rc = _template.from_xml_node(node);
+
+    if (rc == -1)
+    {
+        return -1;
+    }
+
+    increments.init(&_template);
+
+    return 0;
+}
+
+/* -------------------------------------------------------------------------- */
+
+int BackupIncrements::add_increment(const std::string& source, long long size,
+        Increment::Type type)
+{
+    VectorAttribute * va = increments.new_increment(source, size, type);
+
+    if (va == nullptr)
+    {
+        return -1;
+    }
+
+    _template.set(va);
+
+    return 0;
+}
+
+/* -------------------------------------------------------------------------- */
+
+int BackupIncrements::last_increment_id()
+{
+    Increment * li = increments.last_increment();
+
+    if ( li == nullptr )
+    {
+        return -1;
+    }
+
+    return li->id();
+}
+
+/* -------------------------------------------------------------------------- */
+/* -------------------------------------------------------------------------- */
+
+VectorAttribute * IncrementSet::new_increment(const std::string& source, long long sz,
+        Increment::Type type)
+{
+    Increment * li = last_increment();
+
+    int parent = -1;
+
+    if ( li == nullptr )
+    {
+        if ( type != Increment::FULL )
+        {
+            return nullptr;
+        }
+    }
+    else
+    {
+        parent = li->id();
+    }
+
+    int iid = parent + 1;
+
+    VectorAttribute * va = new VectorAttribute("INCREMENT");
+
+    switch (type)
+    {
+        case Increment::FULL:
+            va->replace("TYPE", "FULL");
+            break;
+        case Increment::INCREMENT:
+            va->replace("TYPE", "INCREMENT");
+            break;
+    }
+
+    va->replace("DATE", time(0));
+
+    va->replace("SOURCE", source);
+
+    va->replace("SIZE", sz);
+
+    va->replace("PARENT_ID", parent);
+
+    va->replace("ID", iid);
+
+    add_attribute(attribute_factory(va, iid), iid);
+
+    return va;
+}
+
+/* -------------------------------------------------------------------------- */
+
+long long IncrementSet::total_size()
+{
+    long long sz = 0;
+
+    for (inc_iterator i = begin(); i != end(); ++i)
+    {
+        sz += (*i)->size();
+    }
+
+    return sz;
+}
+
+
+/* -------------------------------------------------------------------------- */
+/* -------------------------------------------------------------------------- */
+
+int BackupIncrements::update_increments(const std::string& incs, const std::string& sz)
+{
+    int first = -1;
+
+    Increment * inc;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We observed that the presence of non-empty statements like `a + b` without clear context or purpose within the codebase can lead to potential confusion or misinterpretation. To mitigate this issue, it&#39;s advisable to ensure that such statements are appropriately integrated into larger constructs that contribute meaningfully to the program&#39;s logic or functionality. This could involve assigning the result of the expression to a variable.